packages/desktop-client/src/components/FinancesApp.js

```diff
@@ -172,78 +226,68 @@ function NavTab({ icon: TabIcon, name, path }) {
 }
 
 function MobileNavTabs() {
+  const { isNarrowWidth } = useResponsive();
   return (
     <div
       style={{
         backgroundColor: 'white',
         borderTop: `1px solid ${colors.n10}`,
         bottom: 0,
         ...styles.shadow,
-        display: 'flex',
+        display: isNarrowWidth ? 'flex' : 'none',
         height: '80px',
         justifyContent: 'space-around',
         paddingTop: 10,
         width: '100%',
       }}
     >
-      <NavTab name="Budget" path="/budget" icon={Wallet} isActive={false} />
-      <NavTab
-        name="Accounts"
-        path="/accounts"
-        icon={PiggyBank}
-        isActive={false}
-      />
-      <NavTab name="Settings" path="/settings" icon={Cog} isActive={false} />
+      <NavTab name="Budget" path="/budget" icon={Wallet} />
+      <NavTab name="Accounts" path="/accounts" icon={PiggyBank} />
+      <NavTab name="Settings" path="/settings" icon={Cog} />
     </div>
   );
 }
 
-class FinancesApp extends Component {
-  constructor(props) {
-    super(props);
-    this.state = { isMobile: isMobile() };
-    this.history = createBrowserHistory();
+function FinancesApp(props) {
+  const patchedHistory = createBrowserHistory();
 
-    let oldPush = this.history.push;
-    this.history.push = (to, state) => {
+  useEffect(() => {
+    let oldPush = patchedHistory.push;
+    patchedHistory.push = (to, state) => {
       let newState = makeLocationState(to.state || state);
       if (typeof to === 'object') {
-        return oldPush.call(this.history, { ...to, state: newState });
+        return oldPush.call(patchedHistory, { ...to, state: newState });
       } else {
-        return oldPush.call(this.history, to, newState);
+        return oldPush.call(patchedHistory, to, newState);
       }
     };
 
     // I'm not sure if this is the best approach but we need this to
     // globally. We could instead move various workflows inside global
     // React components, but that's for another day.
-    window.__history = this.history;
+    window.__history = patchedHistory;
 
     undo.setUndoState('url', window.location.href);
 
-    this.cleanup = this.history.listen(location => {
+    const cleanup = patchedHistory.listen(location => {
       undo.setUndoState('url', window.location.href);
     });
 
-    this.handleWindowResize = this.handleWindowResize.bind(this);
-  }
-
-  handleWindowResize() {
-    this.setState({ isMobile: isMobile() });
-  }
+    return cleanup;
+  }, []);
```

Comment:
If the patch needs to run on every render, the deps array should be removed.

```suggestion
  });
```

Reply:
We patched just once in the class component on initialization, so this should be good.

Reply:
Are you sure that would work? I think the class component used a single instance of the browser history.

Reply:
As a class, it re-created the history with each new budget opened (on my phone, but we use budget as a key, iirc). 
My module var was once for the whole app. So I’m the end, I think your useState suggestion below will be the best match—-only creates history once per budget file.

I’ll give it a shot this weekend. .